### PR TITLE
Attempts to prevent FOUC in .design

### PR DESCRIPTION
**Pull request by brookescarlett**
I did a lot of research to try and understand how to prevent the flash of unstyled content that's currently occurring with .design. Ultimately, I based my solution off of a PR to vercel's Next.js repo where they feature examples of using Next with all different libraries/frameworks (see here: https://github.com/vercel/next.js/pull/20228/files). I'm not convinced that this is 100%, but I do think it's a step in the right direction. I'm finding that it works better on dev than prod (`yarn dev` v. `yarn build && yarn start`) and that it works better in Chrome/Safari than Firefox. 

Let me know what you think and if you have ideas at a better approach here. I know for sure that whatever solution will require modifying the`_document.tsx` file as that's explicitly called out in the `emotion` docs (see here: https://emotion.sh/docs/ssr) but have found a few different implementations out there (especially with the recently released emotion 11)

Very open to feedback/thoughts!

### CODE REVIEW

File: website/pages/_document.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import Document, { Html, Head, Main, NextScript } from 'next/document';
+import * as React from 'react';
+import { renderStatic } from 'utils/renderer';
+
+export default class AppDocument extends Document {
```
Comment (lazytype) [inline website/pages/_document.tsx:5]: can you add docstring with a link to the Next.js emotion example here

File: website/utils/renderer.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import createEmotionServer from '@emotion/server/create-instance';
+import { cache } from '@emotion/css';
+
+export const renderStatic = async (callback: Function) => {
```
Comment (lazytype) [inline website/utils/renderer.ts:4]: `Function` will make `html` be typed as `any`
```suggestion
export const renderStatic = async (callback: () => string) => {
```

**Review by lazytype — APPROVED** [review]

File: website/utils/renderer.ts (hunk 2 above)
Comment (brookescarlett) [reply website/utils/renderer.ts:4]: 👍

File: website/pages/_document.tsx (hunk 1 above)
Comment (brookescarlett) [reply website/pages/_document.tsx:5]: yep

**Pull request merged**